Detectors/TRD/calibration/src/CalibratorVdExB.cxx

```diff
@@ -13,22 +13,160 @@
+    auto xDir = TMath::Cos(impactAngle * TMath::DegToRad());
+    auto yDir = TMath::Sin(impactAngle * TMath::DegToRad());
+    double slope = (TMath::Abs(xDir) < 1e-7) ? 1e7 : yDir / xDir;
+    auto laTan = TMath::Tan(laFit);
+    double lorentzSlope = (TMath::Abs(laTan) < 1e-7) ? 1e7 : 1. / laTan;
+
+    // hit point of incoming track with anode plane
+    double xAnodeHit = trdAnodePlane / slope;
+    double yAnodeHit = trdAnodePlane;
+
+    // hit point at anode plane of Lorentz angle shifted cluster from the entrance -> independent of true drift velocity
+    double xLorentzAnodeHit = trdAnodePlane / lorentzSlope;
+    double yLorentzAnodeHit = trdAnodePlane;
+
+    // cluster location within drift cell of cluster from entrance after drift velocity ratio is applied
+    double xLorentzDriftHit = xLorentzAnodeHit;
+    double yLorentzDriftHit = trdAnodePlane - trdAnodePlane * (vdPreCorr / vdFit);
+
+    // reconstructed hit of first cluster at chamber entrance after pre Lorentz angle correction
+    double xLorentzDriftHitPreCorr = xLorentzAnodeHit - (trdAnodePlane - yLorentzDriftHit) * TMath::Tan(laPreCorr);
+    double yLorentzDriftHitPreCorr = trdAnodePlane - trdAnodePlane * (vdPreCorr / vdFit);
+
+    double impactAngleSim = TMath::ATan2(yAnodeHit, xAnodeHit);
+
+    double deltaXLorentzDriftHit = xAnodeHit - xLorentzDriftHitPreCorr;
+    double deltaYLorentzDriftHit = yAnodeHit - yLorentzDriftHitPreCorr;
+    double impactAngleRec = TMath::ATan2(deltaYLorentzDriftHit, deltaXLorentzDriftHit);
+
+    double deltaAngle = impactAngleRec - impactAngleSim;
+    graph->SetPoint(iPoint, impactAngle, deltaAngle * TMath::RadToDeg());
+    iPoint++;
+  }
+  return graph;
+}
+
+double FitHelper::operator()(const double* par) const
+{
+  double chi2 = 0; // this value is minimized
+  auto graphTmp = createTmpGraph(par[CalibratorVdExB::ParamIndex::VD], par[CalibratorVdExB::ParamIndex::LA]);
+  for (int iBin = 1; iBin <= profiles[currDet]->GetNbinsX(); ++iBin) {
+    auto impactAngle = (profiles[currDet]->GetBinCenter(iBin) + 90) * TMath::DegToRad();
+    auto deltaAlpha = profiles[currDet]->GetBinContent(iBin) * TMath::DegToRad();
+    if (TMath::Abs(deltaAlpha) < 1e-7) {
+      continue;
+    }
+    if (impactAngle < lowerBoundAngleFit || impactAngle > upperBoundAngleFit) {
+      continue;
+    }
+    auto deltaAlphaSim = graphTmp->Eval(impactAngle);
+    chi2 += TMath::Power(deltaAlphaSim - deltaAlpha, 2);
+  }
+  return chi2;
+}
+
 using Slot = o2::calibration::TimeSlot<AngularResidHistos>;
 
 void CalibratorVdExB::initOutput()
 {
   // prepare output objects which will go to CCDB
+  // nothing to be done
+}
+
+void CalibratorVdExB::initProcessing()
+{
+  if (mInitDone) {
+    return;
+  }
+  mFitHelper.lowerBoundAngleFit = 80 * TMath::DegToRad();
+  mFitHelper.upperBoundAngleFit = 100 * TMath::DegToRad();
+  mFitHelper.vdPreCorr = 1.546;
+  mFitHelper.laPreCorr = -0.16133; // OS: what about magnetic field changes?
```

Comment:
this is a temporary fixed value (for simulated tracklets); in the future it will be replaced with vD and LA from CCDB from previous calibration

Reply:
ok thanks then I will put a comment there that these values should eventually be taken from CCDB